### ISSUE TITLE
Since 8.0  update getting following error

### ISSUE THREAD

**Issue by mediaburnwayne** [prof79/fansly-downloader-ng#11]
[16]ERROR | 16:33 ||
Error processing video 'D:\Some\Path\2021-09-28_at_05-28_id_299182134885621760_hash_fad538a4df9a7c6c79ce7c8c546355a9.mp4': Traceback (most recent call last):
  File "fileio\fnmanip.py", line 142, in add_hash_to_other_content
  File "fileio\mp4.py", line 92, i
  
  Running on windows

**Comment by prof79** [comment]
Hi there, can you verify that this is a corrupt MP4 file which didn't download correctly? I should change the error that's bubbling up to be precise and helpful.

That's what line 92 says:

`if file_size < 8:`
`  raise RuntimeError('File is too small.')`

In layman's terms, even the most minimal empty MP4 file would have to have a header of at least 8 bytes, a size specification plus a so-called FourCC code.

So please check:

- Can you double-click/play the offending MP4 in a video player?
- What size does Windows Explorer give/show? (maybe right-click "Properties" and post a screenshot, please)

If it's corrupt as I suspect then I'll change the code to warn about the corrupt file and recommend deleting it so it will get re-downloaded if still available.
Otherwise it gets much more complicated because an internal Python library should return proper file sizes ...

**Comment by vbviaw8467** [comment]
First of all, I would like to express my gratitude for your work.

I have the same symptom since update.
English is not my native language, so there are limitations to my explanations. this is what I figured it out.

1. Downloaded video files after update can play. but all the video files are slightly longer than origin duration time (like 5~30 sec)
Certain file fail to generate a hash and giving a message like mediaburnwayne.
2. So I tested with Avnsx's version and test CRC check. The file before the update matched both versions, but the file after the update did not.
4. I noticed that when downloading a problematic file, the following message displayed.
"pyffmpeg.FFmpeg - INFO - Checking GitHub Activeness: True"

**Comment by prof79** [comment]
@vbviaw8467 Thanks, maybe you should study the release notes a bit more, many things are explained there. We need to differentiate what is normal/expected behavior and what could be a bug or oversight due to the new hashing method:

1. This is `ffmpeg`'s doing by properly mixing the separate streams of a video together instead of patching things up manually. `ffmpeg` and its libraries are written by A/V experts and used all over the place in a myriad of multimedia software, so you and I have to trust that those new files are the (more) correct versions than before. The failing files are the ones that probably need examination.
2. The approach in 0.8.0 is totally different from earlier versions (which used Avnsx's algorithm). This is incomparable starting with the fact that `pyffmpeg` and `PyAV` put different version strings into the files - that alone would break comparison. And that exactly is the reason why I devised a new hashing methodology to make old and new files comparable! Because except the deviating header info those files are in fact the same!
3. This has nothing to do with problematic files or not. This is either shown for the first M3U8 download or maybe even for every M3U8 download, didn't pay that close attention. It only means that `pyffmpeg` is checking for/downloading an adequate `ffmpeg` binary to your machine so the downloaded streams can be converted/merged into an MP4 file. I also find this message unnerving but the way the library is built and Avnsx introduced/used a logger (`loguru`) I could not find a way to suppress this message yet. The hashing code is totally separate from the `ffmpeg` processing.

To sum it up:

1. Concentrate on files where hashing fails. The `fansly_downloader_ng.log` can help you track them down.
2. Do the unhashable files play in a video player?
3. How big are those files (size in bytes) according to Windows Explorer/Nautilus or whatever file manager you're using on your OS?

I have tested the new hashing on an existing collection of ~ 6,425 (!!!) Fansly MP4s and had no error of that kind.

So for me it is very difficult to diagnose this problem from afar. If I had access to a problematic MP4 file things would be so much easier. Aside from that we can only check the basics - primarily the file size and playability of the offending files. They could just simply be broken downloads (server/Internet connection error, whatever).

**Comment by vbviaw8467** [comment]
1. Yes, It can play in a video player perfectly fine.
2. I only have one file that failed to hash. The file size is 5,218,886,392 bytes and size on disk is 5,218,889,728 bytes.

Again, Thank you for all your work and kinldy answering all of my question!

**Comment by vbviaw8467** [comment]
[ [16]ERROR ] [2024-02-02 | 09:34]: 
Error processing video 'path': Traceback (most recent call last):
  File "path\fansly-downloader-ng-main\fileio\fnmanip.py", line 142, in add_hash_to_other_content
    file_hash = hash_mp4file(algorithm, filepath)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "path\fansly-downloader-ng-main\fileio\mp4.py", line 102, in hash_mp4file
    for box in boxes:
  File "path\fansly-downloader-ng-main\fileio\mp4.py", line 44, in get_boxes
    box = MP4Box(
          ^^^^^^^
  File "path\fansly-downloader-ng-main\fileio\mp4.py", line 31, in __init__
    self.fourcc = str(fourcc_bytes, encoding='ascii')
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'ascii' codec can't decode byte 0x83 in position 1: ordinal not in range(128)
----------------------------------------------------------------------------------------------------------
this is my log

**Comment by prof79** [comment]
Thanks! This seems to be indeed different of what the OP wrote.

So either I missed something from a structural side (not an MP4 expert) or there might be something special about that video file.

As I understand it MP4s consist of parts that are called boxes or atoms like a linked list with a length specification and a four-character designation (FourCC) what type of info/data follows within the file. This [FourCC](https://fourcc.org/fourcc.php) is expected to be just English letters and digits but this is not the case at the specific position within your file and that's what the error message says.

Hmm, funny. Wikipedia and some other sources give me the impression that although this should be printable characters only, in theory, they do not always are/have to, it seems. I'll have to do some code changes for that.

It would still be very helpful if you could run a command like this (copy-paste) in `Windows PowerShell` on Windows or `bash` on Linux? (Substitute your MP4 file path/name)

    ~/.pyffmpeg/bin/ffmpeg -v trace -i "path_to_your.mp4" 2>&1 | grep 'type:'

Then please post the results.

**Comment by prof79** [comment]
Btw @mediaburnwayne  and @vbviaw8467 take a look at **v0.8.1** at https://github.com/prof79/fansly-downloader-ng/releases/tag/v0.8.1 whether it fixes your issues.

**Comment by vbviaw8467** [comment]
Sadly, my subscription just expired so I can't check it. Sorry :'(
But I believe the problem has been resolved!

And when I run command on winodws powershell, the following message was displayed.

grep: The term 'grep' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:103
+ ... path.mp4" 2>&1 | grep 'type:'
    + CategoryInfo          : ObjectNotFound: (sed:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

**Comment by prof79** [comment]
@vbviaw8467 Yeah, expired subscription shouldn't matter - because the problem was with reading an already existing/downloaded file!

Sorry, newer PowerShell versions have aliases for UNIX commands so what I posted would have worked independent of platform.

The pure PowerShell/Windows Version would be:

    ~\.pyffmpeg\bin\ffmpeg.exe -v trace -i "path_to_your.mp4" 2>&1 | select-string 'type:'

**Comment by vbviaw8467** [comment]
I tried with updated version. It workd fine! The hash was generated normally without any problem.
Thanks!

I am also attaching the information you mentioned in case it may be of further help.

[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'ftyp' parent:'root' sz: 32 8 5218886392
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'mdat' parent:'root' sz: 5215091127 48 5218886392
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'moov' parent:'root' sz: 3795225 5215091175 5218886392
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'mvhd' parent:'moov' sz: 108 8 3795217
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'trak' parent:'moov' sz: 1684053 116 3795217
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'tkhd' parent:'trak' sz: 92 8 1684045
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'edts' parent:'trak' sz: 48 100 1684045
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'elst' parent:'edts' sz: 40 8 40
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'mdia' parent:'trak' sz: 1683905 148 1684045
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'mdhd' parent:'mdia' sz: 32 8 1683897
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'hdlr' parent:'mdia' sz: 45 40 1683897
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'minf' parent:'mdia' sz: 1683820 85 1683897
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'vmhd' parent:'minf' sz: 20 8 1683812
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'dinf' parent:'minf' sz: 36 28 1683812
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'dref' parent:'dinf' sz: 28 8 28
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stbl' parent:'minf' sz: 1683756 64 1683812
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stsd' parent:'stbl' sz: 216 8 1683748
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'avcC' parent:'stsd' sz: 59 8 114
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'colr' parent:'stsd' sz: 19 67 114
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'pasp' parent:'stsd' sz: 16 86 114
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'btrt' parent:'stsd' sz: 20 102 114
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stts' parent:'stbl' sz: 7576 224 1683748
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stss' parent:'stbl' sz: 5684 7800 1683748
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'ctts' parent:'stbl' sz: 650624 13484 1683748
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stsc' parent:'stbl' sz: 40 664108 1683748
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stsz' parent:'stbl' sz: 339880 664148 1683748
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'co64' parent:'stbl' sz: 679728 1004028 1683748
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'trak' parent:'moov' sz: 2110958 1684169 3795217
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'tkhd' parent:'trak' sz: 92 8 2110950
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'edts' parent:'trak' sz: 36 100 2110950
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'elst' parent:'edts' sz: 28 8 28
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'mdia' parent:'trak' sz: 2110822 136 2110950
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'mdhd' parent:'mdia' sz: 32 8 2110814
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'hdlr' parent:'mdia' sz: 45 40 2110814
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'minf' parent:'mdia' sz: 2110737 85 2110814
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'smhd' parent:'minf' sz: 16 8 2110729
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'dinf' parent:'minf' sz: 36 24 2110729
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'dref' parent:'dinf' sz: 28 8 28
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stbl' parent:'minf' sz: 2110677 60 2110729
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stsd' parent:'stbl' sz: 123 8 2110669
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'esds' parent:'stsd' sz: 51 8 71
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'btrt' parent:'stsd' sz: 20 59 71
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stts' parent:'stbl' sz: 7576 131 2110669
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stsc' parent:'stbl' sz: 892132 7707 2110669
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'stsz' parent:'stbl' sz: 531056 899839 2110669
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'co64' parent:'stbl' sz: 679728 1430895 2110669
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'sgpd' parent:'stbl' sz: 26 2110623 2110669
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'sbgp' parent:'stbl' sz: 28 2110649 2110669
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'udta' parent:'moov' sz: 98 3795127 3795217
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'meta' parent:'udta' sz: 90 8 90
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'hdlr' parent:'meta' sz: 33 8 78
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'ilst' parent:'meta' sz: 45 41 78
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000020a6bc50800] type:'[169]too' parent:'ilst' sz: 37 8 37
[h264 @ 0000020a6bc770c0] nal_unit_type: 7(SPS), nal_ref_idc: 3
[h264 @ 0000020a6bc770c0] nal_unit_type: 8(PPS), nal_ref_idc: 3
[h264 @ 0000020a6bc770c0] nal_unit_type: 7(SPS), nal_ref_idc: 3
[h264 @ 0000020a6bc770c0] nal_unit_type: 8(PPS), nal_ref_idc: 3
[h264 @ 0000020a6bc770c0] nal_unit_type: 9(AUD), nal_ref_idc: 0
[h264 @ 0000020a6bc770c0] nal_unit_type: 7(SPS), nal_ref_idc: 3
[h264 @ 0000020a6bc770c0] nal_unit_type: 8(PPS), nal_ref_idc: 3
[h264 @ 0000020a6bc770c0] nal_unit_type: 6(SEI), nal_ref_idc: 0
[h264 @ 0000020a6bc770c0] nal_unit_type: 5(IDR), nal_ref_idc: 3

**Comment by prof79** [comment]
Thanks, that was extremely helpful! I've just spotted another problem - your file is larger than 4 GiB and so a trick is used to handle larger sizes in an MP4 container. I regrettably did not know this up to this point. As a result my code jumped near the end and got confused. The '[169]too' part, however, is proof that the header parts can contain non-printable characters (169 == byte value outside ASCII).
I need to get a huge MP4 somewhere to see how that special sizing works. Sadly, the official ISO papers are paid stuff and don't come cheap :-( I should take a look whether there are any good existing MP4 libraries for Python ...

**Comment by prof79** [comment]
Thanks to you @vbviaw8467 there is now https://github.com/prof79/fansly-downloader-ng/releases/tag/v0.8.3 😁
